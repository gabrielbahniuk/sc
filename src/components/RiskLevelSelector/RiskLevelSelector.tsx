import { appConfig } from '@/config'
import React from 'react'

type RiskLevelSelector = {
  riskLevel: number
  handleRiskLevel(event: React.ChangeEvent<HTMLSelectElement>): void
}

const RiskLevelSelector: React.FC<RiskLevelSelector> = ({ riskLevel, handleRiskLevel }) => {
  const options = []
  const { MINIMUM_RISK_LEVEL, MAXIMUM_RISK_LEVEL } = appConfig

  for (let i = MINIMUM_RISK_LEVEL; i <= MAXIMUM_RISK_LEVEL; ++i) {
    options.push(
      <option key={i} value={i}>
        {i}
      </option>
    )
  }

  return (
    <select data-cy="riskLevelSelector" onChange={handleRiskLevel} value={riskLevel}>
      {options}
    </select>
  )
}

export { RiskLevelSelector }
