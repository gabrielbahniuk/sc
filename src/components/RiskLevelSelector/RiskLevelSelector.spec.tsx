import React from 'react'
import RiskLevelSelector from '.'
import { create } from 'react-test-renderer'
import { shallow, ShallowWrapper } from 'enzyme'

describe('Test RiskLevelSelector', () => {
  const mockHandleRiskLevel = jest.fn()
  it('should render a risk level selector', () => {
    const riskLevelSelector = create(<RiskLevelSelector riskLevel={3} handleRiskLevel={mockHandleRiskLevel} />).toJSON()
    expect(riskLevelSelector).toMatchSnapshot()
  })
  it('should have 22 html option elements', () => {
    const wrapper: ShallowWrapper = shallow(<RiskLevelSelector riskLevel={3} handleRiskLevel={mockHandleRiskLevel} />)
    expect(wrapper.find('option').length).toEqual(23)
  })
})
