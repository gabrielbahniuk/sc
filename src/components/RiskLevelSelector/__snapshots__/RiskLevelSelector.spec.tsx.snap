// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RiskLevelSelector should render a risk level selector 1`] = `
<select
  data-cy="riskLevelSelector"
  onChange={[MockFunction]}
  value={3}
>
  <option
    value={3}
  >
    3
  </option>
  <option
    value={4}
  >
    4
  </option>
  <option
    value={5}
  >
    5
  </option>
  <option
    value={6}
  >
    6
  </option>
  <option
    value={7}
  >
    7
  </option>
  <option
    value={8}
  >
    8
  </option>
  <option
    value={9}
  >
    9
  </option>
  <option
    value={10}
  >
    10
  </option>
  <option
    value={11}
  >
    11
  </option>
  <option
    value={12}
  >
    12
  </option>
  <option
    value={13}
  >
    13
  </option>
  <option
    value={14}
  >
    14
  </option>
  <option
    value={15}
  >
    15
  </option>
  <option
    value={16}
  >
    16
  </option>
  <option
    value={17}
  >
    17
  </option>
  <option
    value={18}
  >
    18
  </option>
  <option
    value={19}
  >
    19
  </option>
  <option
    value={20}
  >
    20
  </option>
  <option
    value={21}
  >
    21
  </option>
  <option
    value={22}
  >
    22
  </option>
  <option
    value={23}
  >
    23
  </option>
  <option
    value={24}
  >
    24
  </option>
  <option
    value={25}
  >
    25
  </option>
</select>
`;
